### ISSUE TITLE
Leaf decay at physics 2+ eats CPU.

### ISSUE THREAD

**Issue by rdebath** [ClassiCube/MCGalaxy#576]
Checks are added to the list and do not go away.

At physics 1 deleting a wood block touching a leaf block adds items to the check list, after a little while the list empties (once all attached leaves are checked).

For physics 2 or higher the list does not empty, item do get removed from the list but more are continually added.

It appears to be due to the call to `ActivateablePhysics.CheckNeighbours` which adds new checks for attached leaf block for which an old check has been removed.

BTW: Is water supposed to kill trees (leaf blocks) like the small plants, 'cause it doesn't?

``` cs
    public static class LeafPhysics {

        public static void DoLeaf(Level lvl, ref PhysInfo C) {
            if (lvl.physics > 1) { //Adv physics kills flowers and mushroos in water/lava
                ActivateablePhysics.CheckNeighbours(lvl, C.X, C.Y, C.Z);
            }

            // Just immediately remove from physics list
            if (!lvl.Config.LeafDecay) {
                lvl.leaves.Clear();
                C.Data.Data = PhysicsArgs.RemoveFromChecks; return;
            }

            // Delay checking for decay for a random amount of time
            if (C.Data.Data < 5) {
                Random rand = lvl.physRandom;
                if (rand.Next(10) == 0) C.Data.Data++;
                return;
            }
```

**Comment by UnknownShadow200** [comment]
It's intended that water does not kill leaf blocks.

note to self: also remove lvl.leaves as it's useless, replace with stackalloced int array instead

**Comment by UnknownShadow200** [comment]
Should be implemented as of f794a91b44a9853d6be31ce7113cc8efecaa46ce

Still to do tomorrow: Logs should trigger leaf decay for surrounding 4x4x4 region

**Comment by rdebath** [comment]
Looks like you need a `C.Data.Data = PhysicsArgs.RemoveFromChecks;` at the end of `DoLog`
